notebooks/service/service.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb/v2/kit/platform"
+)
+
+// Notebook represents all visual and query data for a notebook.
+type Notebook struct {
+	OrgID     platform.ID  `json:"orgID"`
+	ID        platform.ID  `json:"id"`
+	Name      string       `json:"name"`
+	Spec      NotebookSpec `json:"spec"`
+	CreatedAt time.Time    `json:"createdAt"`
+	UpdatedAt time.Time    `json:"updatedAt"`
+}
+
+// Spec is a specification which is just a blob of content provided by the client.
+type NotebookSpec interface{}
```

Comment:
Is this going to be `interface{}` forever, or is it just a shortcut for now? 😱

Reply:
Hah...well it is just a shortcut for now, but that also implies some grand plans for the future, which isn't exactly the case.

The IDPE version just has it as an arbitrary JSON object. It will look something like the JSON I've currently got in demo data [here](https://github.com/influxdata/influxdb/blob/c8669e80f2470ee633e28ba72cb87c255a6650a7/notebooks/transport/demo_data.go#L52). Some of the keys are *probably* not going to change. The tricky part is the value for the `pipes` key, which is an array of elements which could take on one of a multitude of different shapes.

For now, I'm changing `NotebookSpec` to be a `map[string]interface{}` and leaving it as a separate type, keep the door open for future specification. This is a little bit better than just having it be literally anything, since we know it will at least be an object with string keys. I don't want to over-specify it in comparison to the cloud backend so we can be as flexible as practical in case the frontend starts needing to send different shapes of data.